### PR TITLE
CMR-7107: Add support to bulk update granule S3 URL in UMM-G format.

### PR DESCRIPTION
**Pull request by ygliuvt**

### CODE REVIEW

File: ingest-app/src/cmr/ingest/services/granule_bulk_update/s3/s3_util.clj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  separated by comma, and each url must be started with s3:// (case insensitive).
+  Returns the parsed urls in a list."
+  [input-url]
+  (let [urls (map string/trim (string/split input-url #","))]
+    (doseq [url urls]
+      (when-not (or (string/starts-with? url "s3://")
+                    (string/starts-with? url "S3://"))
+        (errors/throw-service-errors
+         :invalid-data
+         [(str "Invalid URL value, each S3 url must start with s://, but was " url)])))
```
[the remark this answers is not in the document]
Comment (ygliuvt) [reply ingest-app/src/cmr/ingest/services/granule_bulk_update/s3/s3_util.clj:18]: Done.

File: ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj [hunk 2]
```diff
@@ -7,15 +7,23 @@
...
    [cmr.ingest.data.granule-bulk-update :as data-granule-bulk-update]
    [cmr.ingest.data.ingest-events :as ingest-events]
    [cmr.ingest.services.bulk-update-service :as bulk-update-service]
-   [cmr.ingest.services.granule-bulk-update.echo10 :as echo10]
-   [cmr.ingest.services.granule-bulk-update.opendap-util :as opendap-util]
-   [cmr.ingest.services.granule-bulk-update.umm-g :as umm-g]
+   [cmr.ingest.services.granule-bulk-update.opendap.echo10 :as opendap-echo10]
+   [cmr.ingest.services.granule-bulk-update.opendap.opendap-util :as opendap-util]
+   [cmr.ingest.services.granule-bulk-update.opendap.umm-g :as opendap-umm-g]
+   ; [cmr.ingest.services.granule-bulk-update.s3.echo10 :as s3-echo10]
```
[the remark this answers is not in the document]
Comment (ygliuvt) [reply ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj:18]: This is placeholder for the next PR coming later this week.

File: ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  (testing "validate url error scenarios"
+    (are3 [url-value re]
+      (is (thrown-with-msg?
+           Exception re (s3-util/validate-url url-value)))
+
+      "invalid s3 link"
+      "http://example.com/foo"
+      #"Invalid URL value, each S3 url must start with s3://, but was http://example.com/foo"
+      
+      "nvalid s3 link in multiple urls"
```
Comment (mschmele) [inline ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj:21]: Nitpick: spelling
```suggestion
      "invalid s3 link in multiple urls"
```

File: ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      "s3://abc/foo,http://example.com/bar,s3://abc/baz"
```
Comment (mschmele) [inline ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj:22]: We should add a case where the `S` in `s3` is capitalized

**Review by mschmele — APPROVED** [review]

**Review by jaybarra — APPROVED** [review]

File: ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj (hunk 4 above)
Comment (ygliuvt) [reply ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj:22]: I have an active question in the granule bulk update channel to limit the S3 link to be only `s3://`, not `S3://`. I will change one of the s3 to S3 if we decide to allow both.

Comment (ygliuvt) [reply ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj:22]: Actually, the `S3` is already checked on line 9 in the same file. No need to do it here again.

**Pull request merged**